### PR TITLE
fix: Empty connection sets

### LINKED ISSUE #9: Error when connecting empty connection sets
My BSB configuration yielded no connections for the strategy `stellate_to_purkinje`. This resulted in the following error when simulating the circuit with NEST: 

```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/bsb_nest/adapter.py", line 158, in connect_neurons
    connection_model.create_connections(
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/bsb_nest/connection.py", line 68, in create_connections
    for pre_locs, post_locs in self.predict_mem_iterator(
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/bsb_nest/connection.py", line 98, in predict_mem_iterator
    predicted_local_mem = predicted_all_mem / len(cs.get_local_chunks("out"))
ZeroDivisionError: division by zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/cerebellum/cerebellum/test/test_spiking_results.py", line 45, in setUp
    self.simulation_results = self.scaffold.run_simulation("basal_activity")
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/bsb/profiling.py", line 159, in decorated
    return f(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/bsb/core.py", line 444, in run_simulation
    return adapter.simulate(simulation)[0]
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/bsb_nest/adapter.py", line 53, in simulate
    return super().simulate(simulation)
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/bsb/simulation/adapter.py", line 76, in simulate
    data = self.prepare(simulation)
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/bsb_nest/adapter.py", line 68, in prepare
    self.connect_neurons(simulation)
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/bsb_nest/adapter.py", line 163, in connect_neurons
    raise NestConnectError(f"{connection_model} error during connect.")
bsb_nest.exceptions.NestConnectError: <bsb_nest.connection.NestConnection object 'stellate_to_purkinje' at 0x7f6befe1dd60> error during connect.
```
We should add some tests for empty connection sets.

### PR DESCRIPTION
**Pull request by drodarie**
Add checks when connection sets are empty. 

Close #9

### CODE REVIEW

File: bsb_nest/connection.py [hunk 1]
```diff
@@ -95,7 +97,11 @@ def predict_mem_iterator(self, pre_nodes, post_nodes, cs):
         predicted_all_mem = (
             len(pre_nodes) * 8 * 2 + len(post_nodes) * 8 * 2 + len(cs) * 6 * 8 * (16 + 2)
         ) * MPI.get_size()
-        predicted_local_mem = predicted_all_mem / len(cs.get_local_chunks("out"))
+        predicted_local_mem = (
+            (predicted_all_mem / len(cs.get_local_chunks("out")))
+            if len(cs.get_local_chunks("out")) > 0
+            else 0.0
+        )
```
Comment (Helveg) [inline bsb_nest/connection.py]: Iterating over the chunks in an HDF5 file seems incredibly slow. Do it only once:

```suggestion
        n_chunks = len(cs.get_local_chunks("out"))
        predicted_local_mem = (
            (predicted_all_mem / n_chunks)
            if n_chunks > 0
            else 0.0
        )
```

I'm not sure if this applies here, it's just a precaution.

File: tests/test_nest.py [hunk 2]
```diff
@@ -98,6 +99,16 @@ def spy(_):
...
 
+    @patch("bsb_hdf5.connectivity_set.ConnectivitySet.get_local_chunks")
+    def test_empty_conn(self, get_content_mock):
+        # Override get_local_chunks to test empty connection sets
+        get_content_mock.return_value = []
+        cfg = get_test_config("gif_pop_psc_exp")
+        cfg.simulations["test_nest"].connection_models["gif_pop_psc_exp"].rule = None
+        network = Scaffold(cfg, self.storage)
+        network.compile()
+        network.run_simulation("test_nest")
```
Comment (Helveg) [inline tests/test_nest.py:112]: This test doesn't seem to assert anything? Could you add a comment perhaps that it is enough for this test not to fail?

PS: This would make it a "regression test", we could name it `test_regression_issue_9` and use the issue number of the problem that was reported: https://github.com/dbbs-lab/bsb-nest/issues/9

**Review by Helveg — CHANGES_REQUESTED** [review]

**Review by Helveg — APPROVED** [review]
